tools/wasptool

```diff
@@ -258,6 +258,25 @@ def handle_binary_download(c, tname, fname):
     sz = eval(stat)
     bytes_read = 0
 
+    try:
+        if "/" in fname:  # create local directory
+            pile = "./"
+            for local_dir in fname.split("/")[:-1]:
+                if local_dir == ".":
+                    continue
+                try:
+                    os.mkdir(f"{pile}/{local_dir}/")
+                    print(f"Created local dir {pile}/{local_dir}/")
+                except FileExistsError:  # folder already exists
+                    pass
+                pile += local_dir + "/"
+        with open(fname, 'wb') as f:
+            f.write(b"write_test")
+    except:
+        new_name = str(int(time.time())) + ".file"
+        print(f"Error when test writing to {fname}, downloading instead to local file '{new_name}'")
+        fname = new_name
```

Comment:
Technically there could already be a file named like that.

Unique filenames are usually generated using https://docs.python.org/3/library/tempfile.html

Reply:
I am not familiar with tempfile so I'm not comfortable trying this.

While you are correct, this ugly workaroud seems to me vastly better than the current state. Otherwise I could just raise an exception. Your call.